guide/payments/contacts.md

```diff
@@ -0,0 +1,206 @@
+    - file: receive-qr
+      alt: Payment request screen
+      caption: An invoice screen with minimal information.
+    - file: receive-qr-with-contact
+      alt: Payment request screen with a contact assigned
+      caption: The same invoice with a contact assigned.
+imagesActivity:
+    - file: activity
+      alt: Activity screen showing transactions with minimal information
+      caption: By default, payments contain little to no contextual information.
+    - file: activity-annotated
+      alt: Activity screen showing transactions with assigned contacts
+      caption: Assigning contacts to payments makes them much easier to understand and work with.
+    - file: activity-transaction
+      alt: Transaction details
+      caption: Transaction detail screens should have convenient options to assign a contact.
+    - file: sending-pick-contact
+      alt: Screen with a list of contacts to choose from
+      caption: The user picks a contact in a slide-up modal.
+    - file: activity-transaction-annotated
+      alt: Transaction details with assigned contact
+      caption: A payment with an assigned contact.
+
+---
+
+{% include picture.html
+    image = "/assets/images/guide/payments/contacts/contacts-header.jpg"
+    retina = "/assets/images/guide/payments/contacts/contacts-header@2x.jpg"
+    mobile = "/assets/images/guide/payments/contacts/contacts-header-mobile.jpg"
+    mobileRetina = "/assets/images/guide/payments/contacts/contacts-header-mobile@2x.jpg"
+    alt-text = "Illustration of large contact cards"
+    width = 1600
+    height = 400
+    layout = "full-width"
+%}
+
+
+# Contacts
+
+<!--
+
+
+-->
+
+Whether we’re sending emails, physical mail, or following someone on social media, we primarily think in terms of names and faces, and not the respective address or user ID.
+
+Invoices, node IDs and other transaction endpoints in bitcoin and Lightning are highly unintuitive. Abstracting them via a contact list can create a much smoother user experience. There are many [payment request formats]({{ '/guide/payments/request/payment-request-formats/' | relative_url }}), each with unique properties and varying levels of maturity and adoption, requiring unique design solutions. This page uses the more approachable term "address", along with various UI techniques, to abstract these complexities for users.
+
+Let's go over common user interactions around contacts. This will illustrate how such a feature could work, and helps explain the underlying design problems and decisions.
+
+### Adding a contact
+
+The most basic interaction is that a user manually adds a contact by entering their name and an address. This is likely not the most common user flow, as most contacts will be added to incoming payment requests, as illustrated further below. But wallets should generally support manual options, as users may not be able to avoid them.
+
+{% include image-gallery.html pages = page.imagesAddContact %}
```

Comment:
I like the idea of the being able to cross-link the contact metadata with payment details (addresses, as they are referred to on this page). However, there's a lot of stuff I find really strange about this series of example screens.

- xPubs - why can the user import an xPub to a contact? I'm very skeptical that this feature is a good design decision for a mobile LN wallet.
  - **tl;dr** I don't think the majority of mobile users will benefit from storing xPubs with their contacts, and it could actually be a net harm if they mess something up.
  - We all know address re-use has privacy implications. So in theory, you could give your friend your xPub and they can generate fresh addresses to coins to you.
  - But is this a good idea? What if two friends have my xPub? What if two people send to the same address? Are my friends scanning the blockchain to see if any of my addresses have been used already? Are their wallets constantly scanning the blockchain for every single xPub of every contact in their list? (Reductio ad absurdum)
  - Like I brought up on the Jam call this week, I really don't think users should be in the habit of sending the most valuable monetary asset in the entire world to addresses, invoices, etc. that the receiving party has not consented to receiving from.
    - Yes, it's technically possible to send to _any address you want to_, but not good practice, IMO.
  - To me, the most useful situation for things like xPubs for most users are things like BTCPayServer where you want to the instance to generate receive addresses without having your private keys, or maybe something like auto-withdrawals from Swan Bitcoin.
- On-chain addresses
  - Why is there so much focus on on-chain addresses in these screens?
  - You already mentioned in the captions that the address can only be used once. Ok, perfect, I agree, that's a nice feature to restrict the user from re-using an address.
  - But tell me -- if these addresses can only be used once, what user is going to be manually copy and pasting on-chain addresses into their contacts list so they can use those addresses only once?
  - It looks incredibly slick, but copy/pasting individual addresses seems like a really obtuse user experience.
  - I'm not saying no user would ever want that feature -- I just don't see 99% of users doing that, and it seems weird to be featuring that so prominently at the very top of this page.

The ability to cross-link a contacts list with one's on-chain transaction history is very useful. However, I don't really think the ability to manually add on-chain data to a contacts list is a feature that's going to help most users. There's a subtle but important distinction between those two use cases.

Suggested remedy: cut out the manually adding an on-chain address screens or swap it out with a BOLT12 offer or LNURL pay request.

Reply:
Totally understand your rationale and I think some tweaks are necessary. You are viewing these flows very different than what I intended them to be.

Here are some thoughts I had in mind with this:

1. Wallets should be able to identify all address types in existence and have an appropriate response for the user with practical information (even if it has to state that the format is not supported/recommended).
1. We cannot predict the scenarios users find themselves in. There might be really good reasons to save an on-chain address or xpub. I'd rather have the app inform users appropriately (via point 1) and let them go through with it (maybe with a bit of extra friction) than completely restrict it. If we artificially restrict something like this, users may be stuck and won't be able to access their funds at an important moment.
1. Payment request formats are a moving target, so I was more focused on explaining a good UI system to handle various scenarios rather than describing which ones to use for the daily spending use case. Showing two different formats with very different properties helps with that.
1. This first user flow is meant to guide the reader through the most basic elements covered on the page. It is intentionally completely manual and a bit drawn out, because everything else builds on it. Especially with on-chain addresses, the clipboard flow is much more likely to occur. But the idea was to start with the manual basics to avoid instant overload (the first iteration of this page took that approach).

What I'd like to do as a response:

1. Add an example of manually adding an xpub, with a warning below ("You can make repeated payments to this address, but it is highly recommended not to use it. This address type (extended public key) can cause privacy and security problems if not carefully managed"). Maybe add a check-box to let users confirm that they understand the implications. In the end, designers need to consider what their app supports and carefully think through how their app informs users. Showing outlier examples, along with "good" examples helps with that.
2. Add a note below that the app should emphasize Lightning formats for this particular use case (summarizing your considerations above).

How does that sound?

Reply:
OK, did something a little different. I split out the various states for the "add address" interaction into a separate paragraph. This allows the first paragraph to be just about the user flow of contact and address input. The second paragraph goes into the discussion of supporting formats and UX considerations.

I added two states for xpubs. One is a UI that allows their use, but the user needs to toggle an "I understand" box. The second one disallows their use. [Check it here](https://deploy-preview-669--sad-borg-390916.netlify.app/guide/payments/contacts/#supporting-various-payment-request-formats). Also tried to clarify that designers need to consider this for their own apps. It's that middle ground again in the guide between showing the exact solution for a single use case and laying out alternatives for designers to lay out the thought process and explain some of the underlaying problems.

Reply:
@GBKS I love it, thanks for considering my very wordy feedback.

Yeah, I agree with with you that edge cases must be considered. I don't want to argue that apps should _never_ support some of the features you describe in this PR; my primary concern was how it looks presenting it so close to the top of the page. I feel that your new updates have clarified things a lot.